src/Schema/Comparator.php

```diff
@@ -48,16 +52,37 @@ public function __construct(?AbstractPlatform $platform = null)
      * Returns a SchemaDiff object containing the differences between the schemas $fromSchema and $toSchema.
      *
      * This method should be called non-statically since it will be declared as non-static in the next major release.
+     * Its third argument should be itself.
      *
      * @return SchemaDiff
      *
      * @throws SchemaException
      */
     public static function compareSchemas(
         Schema $fromSchema,
-        Schema $toSchema
+        Schema $toSchema /*, self $comparator = null*/
     ) {
-        $comparator       = new self();
+        if (func_num_args() < 3) {
+            Deprecation::trigger(
```

Comment:
I don't think the comparator should report this issue at runtime. Not passing this parameter isn't deprecated. The API consumer will have to pass the third argument if they are affected by the issues. Otherwise, it's OK not to pass it.

Reply:
Are you sure? I'm afraid it might generate support for us (tickets about false-positives, false-negatives etc.). The feature is available, not using it is deprecated IMHO.

Reply:
The way you're describing it makes sense. We do want to warn users that it's better if they use the API by passing an instance but at the same time, the old API is not deprecated. It will look like it does in the next major release.

Semantically, we want to emit an `E_USER_NOTICE` here, not a deprecation message. But I think there's no need to overrotate on this.